### PR TITLE
feat: More useful share button on NFT purchase modal 

### LINKED ISSUE #8378: Bought an NFT - where is share button? + very "useful" QR code
I bought an NFT and got this popup:
Screenshot 2023-12-03 at 16 20 33

Why is there a QR code? What should I do with that QR code? The QR code is useless in this case. Should I scan it and see that, wow - I bought an NFT? Or should I share the QR code? - I need a "share" button instead. So I can share my purchase on Twitter. 😅

Plus this would be cool too - show transaction:
- #8228

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8378

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-12-27 at 13 38 43@2x

https://github.com/kodadot/nft-gallery/assets/44554284/6135e392-7219-4a13-936f-f6ca2e2f4f6f

## Copilot Summary
copilot:summary

copilot:poem

### CODE REVIEW

File: components/MessageNotify.vue [hunk 1]
```diff
@@ -17,20 +17,18 @@
...
+
+        <ShareButton no-shadow />
       </div>
     </div>
   </NeoMessage>
 </template>
 
 <script lang="ts" setup>
+import ShareButton from '@/components/common/ShareButton.vue'
+
```
Comment (roiLeo) [inline components/MessageNotify.vue:29]: please use auto import

File: components/common/ShareButton.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<template>
+  <NeoDropdown position="bottom-left" :mobile-modal="false">
+    <template #trigger="{ active }">
+      <NeoButton
+        :class="isMobileDevice ? 'icon-action' : ''"
```
Comment (roiLeo) [inline components/common/ShareButton.vue:5]: not relevant anymore?

File: components/gallery/GalleryItemButton/GalleryItemShareBtn.vue [hunk 3]
```diff
@@ -1,66 +1,9 @@
 <template>
```
Comment (roiLeo) [inline components/gallery/GalleryItemButton/GalleryItemShareBtn.vue:1]: remove this component if empty, and replace `GalleryItemShareBtn` with `ShareButton`

**Review by roiLeo — COMMENTED** [review]
I would rename `ShareButton` to `ShareButtonDropdown` or `ShareDropdown`

otherwise lgtm

File: components/MessageNotify.vue (hunk 1 above)
Comment (hassnian) [reply components/MessageNotify.vue]: duly noted

File: components/common/ShareButton.vue (hunk 2 above)
Comment (hassnian) [reply components/common/ShareButton.vue]: it's

**Review by roiLeo — APPROVED** [review]
Noice! let's go 🚀 

✅ code lgtm

**Comment by roiLeo** [comment]
@exezbcz is it normal that we added border-radius on this notification modal?

**Comment by exezbcz** [comment]
> @exezbcz is it normal that we added border-radius on this notification modal?

Hey hey, thanks for the tag


Nope, should remain sharp

**Comment by hassnian** [comment]
sharp ✨

CleanShot 2023-12-27 at 16 01 33@2x

**Comment by hassnian** [comment]
@exezbcz should it be sharp for all neo 

> Nope, should remain sharp

everywhere or only this notification , neomessage component had a border-radius: 0.25rem;

File: components/common/ShareDropdown.vue [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+</template>
+
+<script setup lang="ts">
+import {
+  NeoButton,
+  NeoDropdown,
+  NeoDropdownItem,
+  NeoModal,
+} from '@kodadot1/brick'
+import { isMobileDevice } from '@/utils/extension'
```
Comment (daiagi) [inline components/common/ShareDropdown.vue:50]: isMobileDevice checks if it's touch screen (or something similar cant quite remember)

is that truly the intention, or it should be about screen size?

**Comment by prury** [comment]
i think mobile needs some love ❤️, can you check @hassnian ? 


https://github.com/kodadot/nft-gallery/assets/36627808/732c27d8-a175-4033-af2a-2db47e63af51

i cannot be sure if the UI its buggy like the video because i could not ctrl+f5 when switching to mobile view, it always gives me keypair known error, not related i know. 
When i tried on chrome i faced some lagging, responsivity issues.

**Comment by yangwao** [comment]
> i think mobile needs some love ❤️

let's make it in follow up, merging for release 

Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 8.65 USD/DOT ~ 3.468 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0x9c28e2ac60b87c3c38e5d42065f73c134b9c7d8551ae476fe78825aefabebc6c](https://polkadot.subscan.io/extrinsic/0x9c28e2ac60b87c3c38e5d42065f73c134b9c7d8551ae476fe78825aefabebc6c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**